cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -135,6 +135,69 @@ using rhs_iterator = strong_index_iterator<rhs_index_type>;
 
 namespace lexicographic {
 
+/**
+ * @brief Relational comparator functor that evaluates `NaN` as not less, not equivalent, and not
+ * greater than all other values. This is IEE 754 compliant.
+ */
+struct physical_element_comparator {
+  /**
+   * @brief Shared operator for all `Element` types relational comparisons.
+   *
+   * @param[in] lhs first element
+   * @param[in] rhs second element
+   * @return Indicates the relationship between the elements in
+   * the `lhs` and `rhs` columns.
+   */
+  template <typename Element>
+  __device__ constexpr weak_ordering operator()(Element const lhs, Element const rhs) const noexcept
+  {
+    return detail::compare_elements(lhs, rhs);
+  }
+};
+
+/**
+ * @brief Relational comparator functor that evaluates `NaN` as equivalent to other `NaN`s and
+ * greater than all other values.
```

Comment:
```suggestion
 * @brief Relational comparator functor that evaluates `NaN` as equivalent to other `NaN`s and
 *             greater than all other values.
```

Reply:
Sticking with the automated formatting, I hesitate to go against it will force manual changes the next time the comment is modified. e.g. `other 'NaN's       greater than all other values.` is how it would get formatted if the first line of the comment is lengthened.